File: pg_games_api/src/models/playerModel.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IPlayer extends Document {
  name: string;
  role: string; // e.g., Batsman, Bowler, Goalkeeper
  team: mongoose.Types.ObjectId; // Reference to the team
  image?: string; // URL or path to player image
}

const PlayerSchema: Schema = new Schema(
  {
    name: { type: String, required: true },
    role: { type: String, required: true },
    team: { type: mongoose.Schema.Types.ObjectId, ref: 'Team', required: true },
    image: { type: String }, // Player image URL
  },
  { timestamps: true } // Add created_at and updated_at fields
);

const Player = mongoose.models.Player || mongoose.model<IPlayer>('Player', PlayerSchema);

export default Player;

File: pggames/lib/models/player_model.dart
class PlayerModel {
  final String id;
  final String name;
  final String role;
  final String team;
  final String? image;
  final int points;

  PlayerModel({
    required this.id,
    required this.name,
    required this.role,
    required this.team,
    this.image,
    this.points = 0,
  });

  factory PlayerModel.fromJson(Map<String, dynamic> json) => PlayerModel(
    id: json['_id'] ?? '',
    name: json['name'] ?? '',
    role: json['role'] ?? '',
    team: json['team'] ?? '',
    image: json['image'],
    points: json['points'] ?? 0,
  );
}
